Hazards technique for ONIX

gregoriopellegrino opened the pull request:
Thanks to Chris for the work.

I entered the technique of hazards. In order to handle the different permutations of possible hazards, I thought of using an array. I created a "Join array to comma list" function to concatenate the array (thanks to Matt for support).

This is a starting point. As an editor we still have some doubts, which we will dissolve in the call.

madeleinerothberg commented:
This direction makes sense. Bravo for figuring out the array and comma processing.

Should we be brainstorming the right statement for the last case (3. ELSE display "".), which I think is when we don't know if there are hazards or not? Or is that intentionally blank?

gregoriopellegrino commented:
> Should we be brainstorming the right statement for the last case (3. ELSE display "".), which I think is when we don't know if there are hazards or not? Or is that intentionally blank?

While working on the principles, we discussed for a long time what to show in case metadata about hazards is not present. In the end we had decided not to show anything. Hence the idea of leaving the string blank, but we can talk about that in calls.

mattgarrish commented on UX-Guide-Metadata/draft/techniques/onix-metadata/index.html:
```diff
@@ -173,6 +173,32 @@ <h3>Check for Node</h3>
 					<li><b>ELSE</b> return <code>False</code>.</li>
 				</ol>
 			</section>
+			<section id="join-array-to-comma-list">
```
If you join an array with only one element, you get that element back, so only an array with 2 or more hazards will have commas. If you then replace the last comma with " and" you'll get the same result as this function, as it again won't alter the single-hazard case. I'd expect most languages have a "join" function, so this might be more than is needed.

clapierre commented on UX-Guide-Metadata/draft/techniques/onix-metadata/index.html:
```diff
@@ -173,6 +173,32 @@ <h3>Check for Node</h3>
+						<ol class="condition">
+							<li><b>CONCAT</b> <var>string_array[i]</var> <b>TO</b> <var>output_string</var>.</li>
+							<li>
+								<span><b>IF</b> <var>i</var> < (<var>array_length</var> - 2):</span>
```
Extra < present in code

clapierre commented on UX-Guide-Metadata/draft/techniques/onix-metadata/index.html:
```diff
@@ -242,7 +268,86 @@ <h3>Charts, diagrams, and formulas</h3>
+				<ol class="condition">
+					<li>
+						<span><b>IF</b> <var>no_hazards_or_warnings_confirmed</var> <b>OR</b> (<var>no_flashing_hazards</var> <b>AND</b> <var>no_motion_hazards</var> <b>AND</b> <var>no_sound_hazards</var>):</span>
+						<span><b>THEN</b> display <code id="hazards-no">"No hazards"</code>.</span>
```
can we change the id to hazards-none here instead.

gregoriopellegrino commented:
I've made some changes, now I merge in order to move forward, but we can always go back and make changes to hazards section.

Here are the changes made:
- changed the string ID `"hazards none"` (thanks @clapierre )
- refactor of the join array steps, as suggested by @mattgarrish (and removed the function)
- added the singular string for hazards
- added the option in case the publisher specifies that it has no information about hazards
- added a note to say that metadata can be hidden in case metadata is missing
